executor/asyncloaddata/progress.go

```diff
@@ -20,29 +20,58 @@ import (
 	"go.uber.org/atomic"
 )
 
+// LogicalImportProgress is the progress info of the logical import mode.
+type LogicalImportProgress struct {
+	// LoadedFileSize is the size of the data that's loaded in bytes. It's
+	// larger than the actual loaded data size, but due to the fact that reading
+	// is once-a-block and a block may generate multiple tasks that are
+	// concurrently executed, we can't know the actual loaded data size easily.
+	LoadedFileSize *atomic.Int64
+}
+
+// PhysicalImportProgress is the progress info of the physical import mode.
+type PhysicalImportProgress struct {
+	// EncodeFileSize is the size of the file that has finished KV encoding in bytes.
+	// it should equal to SourceFileSize eventually.
+	EncodeFileSize *atomic.Int64
+}
+
 // Progress is the progress of the LOAD DATA task.
 type Progress struct {
 	// SourceFileSize is the size of the source file in bytes. When we can't get
 	// the size of the source file, it will be set to -1.
 	// Currently, the value is read by seek(0, end), when LOAD DATA LOCAL we wrap
 	// SimpleSeekerOnReadCloser on MySQL client connection which doesn't support
 	// it.
-	SourceFileSize int64
-	// LoadedFileSize is the size of the data that will be loaded in bytes. It's
-	// larger than the actual loaded data size, but due to the fact that reading
-	// is once-a-block and a block may generate multiple tasks that are
-	// concurrently executed, we can't know the actual loaded data size easily.
-	LoadedFileSize *atomic.Int64
+	SourceFileSize          int64
+	*LogicalImportProgress  `json:",inline"`
```

Comment:
```suggestion
	LogicalImportProgress  `json:",inline"`
```

Seems can embed with value not pointer?

Reply:
https://github.com/pingcap/tidb/pull/43020/commits/05b7073321626b0b358ea67ee4b76318e4a3574a

inner field pointer is removed